signing_clients/tests/__init__.py

```diff
@@ -254,3 +263,127 @@ def test_metainf_case_sensitivity(self):
         self.assertTrue(ignore_certain_metainf_files('MeTa-InF/MaNiFeSt.Mf'))
         self.assertFalse(ignore_certain_metainf_files('meta-inf/pickles.mf'))
 
+    def test_get_addon_type_from_signature_privileged_webext(self):
+        with open(get_file('webextension_signed.rsa'), 'rb') as f:
+            type = get_addon_type_from_signature(f.read())
+        # Signature occured on Thursday, January 22nd 2015 at 11:02:22am PST
+        # The signing service returns a Python time.time() value multiplied
+        # by 1000 to get a (hopefully) truly unique serial number
+        self.assertEqual(type, 'privileged web extension')
+
+    def test_get_signer_organizational_unit_name(self):
+        with open(get_file('mozilla-generated-by-openssl.pkcs7.der'), 'rb') as f:
+            addon_type = get_signer_organizational_unit_name(f.read())
+
+        self.assertEqual(addon_type, 'Testing')
+
+        with open(get_file('webextension_signed.rsa'), 'rb') as f:
+            addon_type = get_signer_organizational_unit_name(f.read())
+
+        self.assertEqual(addon_type, 'Mozilla Extensions')
+
+        with open(get_file('zigbert.test.pkcs7.der'), 'rb') as f:
+            addon_type = get_signer_organizational_unit_name(f.read())
+
+        self.assertEqual(addon_type, 'Mozilla Addons Dev')
+
+
+class TestSignatureInfo(unittest.TestCase):
+
+    def setUp(self):
+        with open(get_file('mozilla-generated-by-openssl.pkcs7.der'), 'rb') as f:
+            self.pkcs7 = f.read()
+
+        self.info = SignatureInfo(self.pkcs7)
+
+    def test_loading_reading_string(self):
+        info = SignatureInfo(self.pkcs7)
+        self.assertTrue(isinstance(info.content, collections.OrderedDict))
+
+    def test_loading_pass_signature_info_instance(self):
+        info = SignatureInfo(self.pkcs7)
+        self.assertTrue(isinstance(info.content, collections.OrderedDict))
+
+        info2 = SignatureInfo(info)
+
+        self.assertTrue(isinstance(info2.content, collections.OrderedDict))
+        self.assertEqual(info2.content, info.content)
+
+    def test_issuer_serial_number(self):
+        self.assertEqual(self.info.issuer_serial_number, 1498181554500)
+
+    def test_issuer(self):
+        self.assertEqual(self.info.issuer, collections.OrderedDict([
+            ('country_name', 'US'),
+            ('state_or_province_name', 'CA'),
+            ('locality_name', 'Mountain View'),
+            ('organization_name', 'Addons Test Signing'),
+            ('common_name', 'test.addons.signing.root.ca'),
+            ('email_address', 'opsec+stagerootaddons@mozilla.com')
+        ]))
+
+    def test_signer_certificate(self):
```

Comment:
If you have them lying around somewhere, absolutely! I filed this as a follow-up but the more test-data we can get the better.